### PR TITLE
feat(release) Add craft descriptor and release workflow

### PR DESCRIPTION
**Pull request by fpacifici**
This is supposed to set up releasing for this project according to https://github.com/getsentry/craft#integrating-your-project-with-craft.

Specifically it adds the release GHA workflow and the craft descriptor file that publish the binary generated by the `build_library` workflow in the sentry registry.

### CODE REVIEW

**Review by evanh — COMMENTED** [review]
I think you also need a `bump_version.sh` script. The release action calls that script to update the version number in your codebase before committing. See https://github.com/getsentry/snuba-sdk/blob/main/scripts/bump-version.sh

**Comment by fpacifici** [comment]
I don't think the version is stored anywhere in the code base. Is it actually needed ?

**Review by BYK — CHANGES_REQUESTED** [review]

File: .craft.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# TODO: Add a changelog
+changelogPolicy: none
+
+artifactProvider:
+  name: github
+statusProvider:
+  name: github
+
+targets:
+  - name: github
```
[the remark this answers is not in the document]
Comment (fpacifici) [reply .craft.yml:17]: fixed

File: .craft.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - name: registry
+    type: app
+    urlTemplate: "https://downloads.sentry-cdn.com/wal2json/{{version}}/{{file}}"
+    includeNames: /^wal2json-Linux-x86_64.so$/
+    config:
+      canonical: "app:relay"
```
[the remark this answers is not in the document]
Comment (fpacifici) [reply .craft.yml]: Well, I thought publishing to the registry was required. It seemed so from the docs. But you are right, it is enough for the file to be downloadable. 

Should I just remove the registry target entirely ?

[the remark this answers is not in the document]
Comment (BYK) [reply .craft.yml]: >I removed the entire target. Is this the right way?

Yup, let's go without the registry for now.

**Review by BYK — APPROVED** [review]

**Pull request merged**